File: break-a-palindrome/break-a-palindrome.cpp
class Solution {
public:
    string breakPalindrome(string palindrome) {
        if (palindrome.size() == 1) {
            return "";
        }
        for (int i = 0; i < palindrome.size() / 2; ++ i) {
            if (palindrome[i] != 'a') {
                return palindrome.substr(0, i) + 'a' + 
                    palindrome.substr(i + 1);
                // or the following should work as well
                // palindrome[i] = 'a';
                // return palindrome;
            }
        }
        return palindrome.substr(0, palindrome.size() - 1) + 'b';
        // or the following should work
        // palindrome[palindrome.size() - 1] = 'b';
        // return palindrome;
    }
};

File: break-a-palindrome/break-a-palindrome.java
class Solution {
    public String breakPalindrome(String palindrome) {
        if (palindrome.length() == 1) return "";
        for (int i = 0; i < palindrome.length() / 2; i++) {
            if (palindrome.charAt(i) != 'a') {
                return palindrome.substring(0, i) + 'a' +
                    palindrome.substring(i + 1);
            }
        }
        return palindrome.substring(0, palindrome.length() - 1) + 'b';
    }
}